pages/api/user/index.ts

```diff
@@ -0,0 +1,178 @@
+import { PrismaClient } from '@prisma/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, EmptyPayload, StatusMessageType } from '../../../types/apiResponse';
+import { AccountPostData } from '../../../types/user';
+import { Nullable } from '../../../types/utils';
+import { getUserFromJwt, UserDetailsFromRequest, withAuthUser } from '../../../utils/auth/jwtHelpers';
+import { createJsonResponse, HttpMethod, HttpStatus, rejectHttpMethod } from '../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../utils/prisma/prismaHelpers';
+
+const prisma = new PrismaClient();
+
+/** Limited set of data representing user and the list of user application IDs. Only for use in this endpoint. */
+interface UserData {
+  uid: string;
+}
+
+enum MessageType {
+  // POST link accounts.
+  NEW_USER_LINKED,
+  NEW_USER_UNVERIFIED,
+  OLD_USER_NOT_FOUND,
+  NEW_USER_EXISTS,
+  INVALID_OLD_TOKEN,
+  OLD_USER_VERIFIED,
+  // POST account creation only.
+  USER_CREATED,
+  USER_ALREADY_EXISTS,
+  // DELETE accounts
+  USER_DELETED,
+  DELETE_TARGET_NOT_FOUND,
+}
+
+const messages = Object.freeze({
+  // POST messages (without oldToken)
+  [MessageType.USER_CREATED]: { type: StatusMessageType.SUCCESS, message: 'New user with UID added.' },
+  [MessageType.USER_ALREADY_EXISTS]: { type: StatusMessageType.ERROR, message: 'User with UID already exists.' },
+  // POST messages (with oldToken)
+  [MessageType.NEW_USER_LINKED]: {
+    type: StatusMessageType.SUCCESS,
+    message: 'Old UID has been replaced with new UID.',
+  },
+  [MessageType.NEW_USER_UNVERIFIED]: {
+    type: StatusMessageType.ERROR,
+    message: 'An unverified account cannot be the target of link.',
+  },
+  [MessageType.INVALID_OLD_TOKEN]: {
+    type: StatusMessageType.ERROR,
+    message: 'Could not validate and decode "oldToken".',
+  },
+  [MessageType.OLD_USER_VERIFIED]: { type: StatusMessageType.ERROR, message: 'Old UID is already linked.' },
+  [MessageType.NEW_USER_EXISTS]: {
+    type: StatusMessageType.ERROR,
+    message: 'New UID already exists and cannot be linked.',
+  },
+  [MessageType.OLD_USER_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Old UID not found.' },
+  // DELETE messages.
+  [MessageType.USER_DELETED]: { type: StatusMessageType.SUCCESS, message: 'Deleted UID and their data.' },
+  [MessageType.DELETE_TARGET_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Could not find UID to delete.' },
+});
+
+async function handler(currentUid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<UserData>>) {
+  switch (req.method) {
+    case HttpMethod.POST:
+      return handlePost(currentUid, req, res);
+    case HttpMethod.DELETE:
```

Comment:
Lets delete this DELETE endpoint for how. I don't see a use case for this now.

Reply:
Resolved: we are keeping DELETE